plugins/modules/dynamic_group.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2024, Network to Code (@networktocode) <info@networktocode.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: dynamic_group
+short_description: Creates or removes dynamic group from Nautobot
+description:
+  - Creates or removes dynamic group from Nautobot
+notes:
+  - Tags should be defined as a YAML list
+  - This should be ran with connection C(local) and hosts C(localhost)
+author:
+  - Travis Smith (@tsm1th)
+requirements:
+  - pynautobot
+version_added: "5.5.0"
+extends_documentation_fragment:
+  - networktocode.nautobot.fragments.base
+  - networktocode.nautobot.fragments.tags
+  - networktocode.nautobot.fragments.custom_fields
+options:
+  name:
+    description:
+      - The name of the dynamic group
+    required: true
+    type: str
+  description:
+    description:
+      - The description of the dynamic group
+    required: false
+    type: str
+  group_type:
+    description:
+      - Required if I(state=present) and the dynamic group does not exist yet
+      - The group type of the dynamic group (dynamic-filter|dynamic-set|static)
```

Comment:
```suggestion
    choices: [ dynamic-filter, dynamic-set, static ]
```

Might as well make this explicit rather than keeping it free-form.